src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -5,20 +5,116 @@
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int buckets;
+    private Node[] table;
+
+    public MyHashMap() {
+        this.table = new Node[INITIAL_CAPACITY];
+        this.buckets = 0;
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        addNode(key, value, table);;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
+            return null;
+        }
+        Node searchNode = table[indexFor(key, table.length)];
+        while (true) {
+            if (areKeysEqual(key, (K) searchNode.key)) {
+                return (V) searchNode.value;
+            }
+            if (searchNode.next == null) {
+                return null;
+            }
+            searchNode = searchNode.next;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addNode(K key, V value, Node[] table) {
```

Comment:
Don't we have access to table anyway?

Reply:
@dmytro-hryhoruk 
Method addNode() is used when something is added in map standard way and also when resizing is performed.
So I pass table parameter to choose where node should be written - into existing table or new (resized) one.

Reply:
you can save the old array to local variable and reassign table = new Node[table.length * 2]. So You don't need to pass any table into your method

Reply:
done